### ISSUE TITLE
Deploy this service throw Ingress GKe: Error UNHEALTHY service

### ISSUE THREAD

**Issue by masalinas** [eclipse/mosquitto#1646]
I tried to deploy a Pod with this docker image inside Google Cloud Platform and expose the port 9001 Websockets throw ingress GKE.

I have a problem because kubernetes told me that the service is UNHEALTHY because the load balancer must send a health probe against any service. But this is not a HTTP service is a TCP service, so How could add a health probe of this service to expose the 9001 port throw ingress GKE??

**Comment by ralight** [comment]
I'm sorry, I really don't know. Doing some searching there are people using Mosquitto with GCP and kubernetes successfully, so it must be possible but I can't help you myself.

**Comment by masalinas** [comment]
Please could you send me some of these links, because I don’t find anything

**Comment by ralight** [comment]
This was the most promising looking: https://cloud.google.com/community/tutorials/kube-mqtt

I freely admit it may be completely different to what you were wanting to do, it really was just a quick search.

**Comment by masalinas** [comment]
Yes I read this post before, but it’s use the MQTT service of GCP not
mosquitto. My problem is with the 9001 Websocket because the LoadBalancer
of GCP set my Mosquitto service like UNHEALTH, because Ingress of GCP
always create a health check of type HTTP and it must be TCP. Before
deployed I change this type to TCP and works during a minute but suddenly
later GPC set again the health check to HTTP so again the connection to
9001 is closed is crazy
.
